source/schemas/types.txt

```diff
@@ -509,3 +509,78 @@ with the ``title`` field set to the embedded object's type name:
             }
       }
    }
+
+.. _geospatial
+
+
+Geospatial Data
+---------------
+
+Geospatial data describes points and other data on the earth's surface. App 
+Services supports defining GeoJSON Point data in a schema. 
+
+GeoJSON Point
+~~~~~~~~~~~~~
+
+A GeoJSON Point is a single location on the Earth's surface, described by its 
+longitude and latitude. To ensure wide compatibility, App Services stores these 
+points in accordance to the `GeoJSON spec <https://datatracker.ietf.org/doc/html/rfc7946>`_, 
+which has additional requirements. Specifically, you must apply the following 
+rules to the GeoPoint:
+
+- It must be of type ``object``
+- It must be embedded within another type
+- It must contain the following fields:
+
+  - ``type``, of type ``string``, which is always set to "Point"
+  - ``coordinates``, of type ``array``. The ``coordinates`` array contains 
+    exactly 2 double values. The first double is the longitude of the point and 
+    the second is the latitude.
+
+- Additional properties are allowed.
+ 
+ The following is the schema of a GeoPoint:
+
+.. code-block:: json
+
+   {
+     "title": "MyObject",
+     "properties": {
+       "location": {
+         "bsonType": "object",
+         "required": [ "type" ],
+         "properties": {
+           "type": {
+             "bsonType": "string",
+           },
+           "coordinates": {
+             "bsonType": "array",
+             "items": {
+               "bsonType": "double"
+             }
+           }
+         }
+       },
+       "another_property": {
+             "bsonType": "double"
+       },
+     }
+   }
+
+Querying GeoJSON Points
+~~~~~~~~~~~~~~~~~~~~~~~
+
+You can query geospatial points when using :ref:`Flexible Sync <flexible-sync>`. 
+No additional configuration is required. 
+
+To learn about querying against geospatial data, see:
+
+- :ref:`Geospatial - .NET SDK <dotnet-geospatial>`
+- :ref:`Geospatial - Kotlin SDK <kotlin-geospatial>`
+- :ref:`Geospatial - Node.js SDK <node-data-types-geospatial>`
+- :ref:`Geospatial - React Native SDK <react-native-data-types-geospatial>`
+
+.. note::
+
+   Atlas does not support creating an index on Geospatial fields, so querying 
+   large collections may reduce performance.
```

Comment:
This just sounds... wrong? Can I not create a 2dsphere index in Atlas?

Reply:
https://jira.mongodb.org/browse/SERVER-80402

Reply:
Yeah, the limitation here is that you cannot create a 2dsphere index on an _array_ of Point objects. Flexible sync will index your query in our internal metadata under the hood and because setting that index is unsupported in MongoDB for an array of Point objects, those will be unindexed queries which would result in degraded performance.

Reply:
I'm still a bit confused by the intricacies here. You can define a 2dsphere index on documents that contain a single GeoPoint (as per [this doc](https://www.mongodb.com/docs/manual/core/indexes/index-types/geospatial/2dsphere/create/)). But Flexible Sync...does what?...that causes those queries to be un-indexed?

Reply:
Yeah, for queries on an array of Point objects, they will be unindexed; however, for queries on a single Point object, they will be indexed with the 2dsphere index.